scipy/special/cdf_wrappers.c

```diff
@@ -87,27 +98,37 @@ extern void F_FUNC(cdfbin,CDFBIN)(int*,double*,double*,double*,double*,double*,d
 
 double cdfbin2_wrap(double p, double xn, double pr) {
   int which=2;
-  double q=1.0-p, s, ompr=1.0-pr, bound;
-  int status;  
+  double q=1.0-p, s=0, ompr=1.0-pr, bound=0;
+  int status=5;
   
+  if (npy_isnan(p) || npy_isnan(q) || npy_isnan(s) || npy_isnan(xn) || npy_isnan(pr) ||
+      npy_isnan(ompr) || npy_isnan(bound)) {
+      return NPY_NAN;
+  }
+
   F_FUNC(cdfbin,CDFBIN)(&which, &p, &q, &s, &xn, &pr, &ompr, &status, &bound);
   if (status) {
     show_error("cdfbin2", status, bound);
-    if ((status < 0) || (status==3) || (status==4)) return (NPY_NAN);
+    if ((status < 0) || (status==3) || (status==4) || (status==10)) return (NPY_NAN);
     if ((status == 1) || (status == 2)) return bound;
   }
   return s;
 }
 
 double cdfbin3_wrap(double s, double p, double pr) {
   int which=3;
-  double q=1.0-p, xn, ompr=1.0-pr, bound;
-  int status;  
+  double q=1.0-p, xn=0, ompr=1.0-pr, bound=0;
+  int status=5;
+
+  if (npy_isnan(p) || npy_isnan(q) || npy_isnan(s) || npy_isnan(xn) || npy_isnan(pr) ||
+      npy_isnan(ompr) || npy_isnan(bound)) {
+      return NPY_NAN;
+  }
 
   F_FUNC(cdfbin,CDFBIN)(&which, &p, &q, &s, &xn, &pr, &ompr, &status, &bound); 
   if (status) {
     show_error("cdfbin3", status, bound);
-    if ((status < 0) || (status==3) || (status==4)) return (NPY_NAN);
+    if ((status < 0) || (status==3) || (status==4) || (status==10)) return (NPY_NAN);
```

Comment:
``status==5`` not 10? The first couple of changes had ``int status=10`` and a check ``status==10``, then you changed to ``int status=5`` but still check for 10 (multiple times more in this file).